src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -799,45 +799,31 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
                     index++;
                 }
             }
-
-            
-            clusterConnections.ForEach(connection =>
+            using (var undoGroup = wsViewModel.Model.UndoRecorder.BeginActionGroup())
             {
-                // Connect the nodes
-                var sourceNode = clusterMapping[connection.StartNode.NodeId];
-                var targetNode = clusterMapping[connection.EndNode.NodeId];
-                // The port index is 1- based (currently a hack and not expected from service)
-                var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.PortModel.Index == connection.StartNode.PortIndex - 1);
-                var targetPort = targetNode.InPorts.FirstOrDefault(p => p.PortModel.Index == connection.EndNode.PortIndex - 1);
-                var commands = new List<DynamoModel.ModelBasedRecordableCommand>
-                    {
-                        new DynamoModel.MakeConnectionCommand(sourceNode.Id.ToString(), connection.StartNode.PortIndex - 1, PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin),
-                        new DynamoModel.MakeConnectionCommand(targetNode.Id.ToString(), connection.EndNode.PortIndex - 1, PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End),
-                    };
-                commands.ForEach(c =>
+                clusterConnections.ForEach(connection =>
                 {
-                    try
+                    // Connect the nodes    
+                    var sourceNode = clusterMapping[connection.StartNode.NodeId].NodeModel;
+                    var targetNode = clusterMapping[connection.EndNode.NodeId].NodeModel;
+                    // The port index is 1- based (currently a hack and not expected from service)
+                    var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.Index == connection.StartNode.PortIndex - 1);
+                    var targetPort = targetNode.InPorts.FirstOrDefault(p => p.Index == connection.EndNode.PortIndex - 1);
+
+                    var connector = ConnectorModel.Make(sourceNode, targetNode, connection.StartNode.PortIndex - 1, connection.EndNode.PortIndex - 1);
+                    if (connector != null && !targetPort.IsConnected)
```

Comment:
@johnpierson I added some `IsConnected` check here and below, PTAL

Reply:
That still doesn't seem to be working in some cases. And I am still getting a crash if I try to delete that node with 2 wires in 1 input port.
![image](https://github.com/user-attachments/assets/8fc7f5bb-bf2a-41a9-afab-0ffa95406b49)